File: RainIt/RainIt.Domain/Repository/RoutinePattern.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RainIt.Domain.Repository
{
    public class RoutinePattern
    {
        public int RoutinePatternId { get; set; }
        public int? RoutineId { get; set; }
        public virtual Routine Routine { get; set; }
        public int? PatternId { get; set; }
        public virtual Pattern Pattern { get; set; }
        public int UserId { get; set; }
        public virtual User User { get; set; }
    }
}

File: RainIt/rain_it/rain_it/domain/routine_pattern.py
'''
This is the routine pattern response object 
'''
from domain.pattern import Pattern


class RoutinePattern(object):    
    def __init__(self, routine_pattern_id, pattern, repetitions):
        self.routine_pattern_id = int(routine_pattern_id)
        self.repetitions = int(repetitions)
        if not "Matrix" in pattern:
            pattern["Matrix"] = []
        self.pattern = Pattern(pattern["PatternId"], pattern["Path"], pattern["ConversionParameterDTO"], pattern["Matrix"])
    
    def get_as_dictionary(self):
        return {"RoutinePatternId":self.routine_pattern_id, "PatternDTO":self.pattern.get_as_dictionary(), "Repetitions":self.repetitions}
    
    def __eq__(self, other): 
        return self.routine_pattern_id == other.routine_pattern_id and self.pattern == other.pattern and self.repetitions == other.repetitions